femtools/Particles.F90

```diff
@@ -946,8 +1067,110 @@ subroutine move_particles(state, dt, timestep)
       call move_lagrangian_detectors(state, particle_lists(i), dt, &
            particle_lists(i)%total_attributes)
     end do
+
+    call profiler_toc("particle_advection")
   end subroutine move_particles
 
+  !> Initialise constant attribute values before diagnostic fields are set
+  subroutine initialise_constant_particle_attributes(state, subgroup_path, p_list)
```

Comment:
It is a little bit of a shame to have to duplicate just this constant-value part out of the attribute update, but I can't really think of a better solution at the moment...